### PR TITLE
Stop large look changes from crashing the server

### PR DESCRIPTION
**Pull request by Jaren8r**
Supersedes #7459, with a proper diff this time

### CODE REVIEW

**Comment by Lulu13022002** [comment]
This can also happens for all projectile see the Projectile#lerpRotation. But since this is just faster/simpler than looping over a large value, you should search in the code all bad pattern of this.

**Comment by Jaren8r** [comment]
That's the only other use of that pattern I could find in the server code, so I went ahead and changed that too.

**Comment by Lulu13022002** [comment]
Hmm what about ProjectileUtil#rotateTowardsMovement (will probably not crash) and Mth#rotlerp then? I can't test if these crash because of the two other loop that run in the tick directly. Also the fix provided for the projectile is wrong, you forgot to increment instead of replace
basically: prevRot **+**= Math.round((newRot - prevRot) / 360.0F) * 360.0F;

**Comment by Jaren8r** [comment]
> Also the fix provided for the projectile is wrong, you forgot to increment instead of replace

Blah, good catch. It's fixed now.

File: patches/server/0936-Stop-large-look-changes-from-crashing-the-server.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jaren Knodel <Jaren@Knodel.com>
```
Comment (kennytv) [inline patches/server/0936-Stop-large-look-changes-from-crashing-the-server.patch]: Put the original patch author here

**Review by kennytv — CHANGES_REQUESTED** [review]

File: patches/server/0936-Stop-large-look-changes-from-crashing-the-server.patch (hunk 1 above)
Comment (Jaren8r) [reply patches/server/0936-Stop-large-look-changes-from-crashing-the-server.patch]: Done

**Review by kennytv — APPROVED** [review]

**Comment by asquared31415** [comment]
Can this get backported to all supported versions? It affects 1.18.2 and possibly other versions as well.

**Comment by electronicboy** [comment]
1.18 is EOL and has far exceeded its support lifecycle, if somebody wants to open a PR, maybe; but, I can't justify throwing any time towards it

**Pull request merged**